x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/annotations/Annotation.java

```diff
@@ -80,11 +71,17 @@ public String toString() {
     public static final ParseField BY_FIELD_NAME = new ParseField("by_field_name");
     public static final ParseField BY_FIELD_VALUE = new ParseField("by_field_value");
 
+    /**
+     * Parses {@link Annotation} using a strict parser.
+     */
     public static Annotation fromXContent(XContentParser parser, Void context) {
         return PARSER.apply(parser, context).build();
     }
 
-    private static final ObjectParser<Builder, Void> PARSER = new ObjectParser<>(ANNOTATION.getPreferredName(), true, Builder::new);
+    /**
+     * Strict parser for cases when {@link Annotation} is returned from C++ as an ML result.
+     */
+    private static final ObjectParser<Builder, Void> PARSER = new ObjectParser<>(ANNOTATION.getPreferredName(), false, Builder::new);
```

Comment:
I think it would be worth naming the variable `STRICT_PARSER`.  It someone ever does need to parse these documents out of an index then a lenient parser will be needed, but at that time it would be easy to assume that a variable named simply `PARSER` can be used.  The person doing that future work won't read the discussion in this PR - they'll just go by what they can easily navigate to in the IDE.

Reply:
Good point.

Done.